src/app/clusters/operational-credentials-server/operational-credentials-server.cpp

```diff
@@ -648,12 +648,13 @@ bool emberAfOperationalCredentialsClusterAddNOCCallback(app::CommandHandler * co
     err = groups->SetKeySet(fabricIndex, compressed_fabric_id, keyset);
     VerifyOrExit(err == CHIP_NO_ERROR, nocResponse = ConvertToNOCResponseStatus(err));
 
-    // We might have a new operational identity, so we should start advertising it right away.
-    app::DnssdServer::Instance().AdvertiseOperational();
-
     // The Fabric Index associated with the armed fail-safe context SHALL be updated to match the Fabric
     // Index just allocated.
-    failSafeContext.SetAddNocCommandInvoked(fabricIndex);
+    err = failSafeContext.SetAddNocCommandInvoked(fabricIndex);
+    VerifyOrExit(err == CHIP_NO_ERROR, nocResponse = ConvertToNOCResponseStatus(err));
```

Comment:
This will get to 
```
exit:
    gFabricBeingCommissioned.Reset();
    SendNOCResponse(commandObj, commandPath, nocResponse, fabricIndex, CharSpan());
```
    
 The currently rolling back not enough?

Reply:
Acked